### ISSUE TITLE
datazilla_client should enumerate posting errors

### ISSUE THREAD

**Issue by k0s** [mozilla/datazilla_client#21]
When posting, there are a number of things that could go wrong:
- host does not exist, the attempt to connect fails
- the project chosen does not exist (is this an error? or should the project just be created?)
- the credentials are wrong
- malformed data 
- other server error
- ??? (not sure what else could go wrong?)

DatazillaRequest should make it easy to determine what is going wrong. I would probably return the error or None on a successful post

**Comment by k0s** [comment]
we should also have tests for these :/  Will not be fun to write

**Comment by malini** [comment]
Does the server respond with appropriate errors for each case (wrong credentials, project does not exist, etc.)?

**Comment by k0s** [comment]
I don't know what the current state is...but if not, it should.

:jeads?

**Comment by carljm** [comment]
Yes, I think all of those error conditions are currently reported clearly by the server, though they are reported via a human-friendly status string in a tiny JSON wrapper, not with any machine-interpretable error code beyond the HTTP status code (which is generally appropriate).

So the client could just pass on the error text from the server in case of a JSON response with a non-200 HTTP code, rather than trying to interpret it.

Tests shouldn't be too bad, will just require a bit of urllib monkeypatching (which at least one test already does).
